documentation/tutorials/get-started-with-phoenix.md

```diff
@@ -0,0 +1,506 @@
+
+13:57:23.163 [info] == Migrated 20230208045100 in 0.0s
+
+13:57:23.213 [info] == Running 20230208045101 MyAshPhoenixApp.Repo.Migrations.InitialMigration.up/0 forward
+
+13:57:23.213 [info] create table posts
+
+13:57:23.220 [info] == Migrated 20230208045101 in 0.0s
+```
+
+If your terminal looks like this then you have successfully created your first ash postgres resource.
+
+### Interacting with your resource
+
+All interaction with your resource attributes **ALWAYS** occur through an **action**. In our resource we are using the default actions for `:create, :read, :update, :destroy`. Create and update actions **ALWAYS** take a changeset. Ash changesets are conceptually similar to [Ecto changesets](https://hexdocs.pm/ecto/Ecto.Changeset.html). They're data structures which represent an intended change to an Ash resource.
+
+Lets write a test to show how to interact with our resource.
+
+```elixir
+defmodule MyAshPhoenixApp.Blog.PostTest do
+  use MyAshPhoenixApp.DataCase, async: true
+
+  test "testing blog post actions" do
+    ### CREATE ACTION - create new blog post ###
+    # We create an Ash changeset we intent to use to create an Ash resource
+    create_post_changeset =
+      Ash.Changeset.for_create(
+        # We specify which resource we want to create a changeset for
+        MyAshPhoenixApp.Blog.Post,
+        # We name the specific create action
+        :create,
+        # We pass the attributes we want to initialise our resource with
+        %{title: "hello world"}
+      )
+
+    # will return:
+    #
+    # #Ash.Changeset<
+    #   action_type: :create,
+    #   action: :create,
+    #   attributes: %{title: "hello world"},
+    #   relationships: %{},
+    #   errors: [],
+    #   data: #MyAshPhoenixApp.Blog.Post<
+    #     __meta__: #Ecto.Schema.Metadata<:built, "posts">,
+    #     id: nil,
+    #     title: nil,
+    #     content: nil,
+    #     aggregates: %{},
+    #     calculations: %{},
+    #     __order__: nil,
+    #     ...
+    #   >,
+    #   valid?: true
+    # >
+
+    # This changeset is given to the Ash Api the resource belongs to (Blog in our case).
+    # The Api then tries to create the resource specified in the changeset.
+    assert %{title: "hello world"} = MyAshPhoenixApp.Blog.create!(create_post_changeset)
+    # will return:
+    #
+    # #MyAshPhoenixApp.Blog.Post<
+    #   __meta__: #Ecto.Schema.Metadata<:loaded, "posts">,
+    #   id: "d70dd979-0b30-4a3f-beb2-2d5bb2e24af7",
+    #   title: "hello world",
+    #   content: nil,
+    #   aggregates: %{},
+    #   calculations: %{},
+    #   __order__: nil,
+    #   ...
+    # >
+
+    ...
+```
+
+Now lets read all of the data in the resource to check the creation we just performed worked. Notice how we don't need changeset here as we are not changing the data, we're just reading it.
+
+```elixir
+  ...
+
+  ### READ ACTION - read blog post(s) ###
+    assert [first_post = %{title: "hello world"}] =
```

Comment:
Is the `first_post =` necessary?

Reply:
yea it is, you need to assign the resource to a variable because its used later in the `:update` action example.

Reply:
addressed: [5c2cb7f](https://github.com/ash-project/ash/pull/504/commits/5c2cb7fdf6732e6b12681b9cd23819322f0baa87)